blocks/v2-hero/v2-hero.js

```diff
@@ -0,0 +1,98 @@
+import {
+  createElement,
+  isVideoLink,
+  removeEmptyTags,
+} from '../../scripts/scripts.js';
+
+const checkVideoTime = (event) => {
+  const video = event.target;
+  if (video.currentTime > 3) {
+    video.pause();
+    video.removeEventListener('timeupdate', checkVideoTime);
+  }
+};
+
+export default async function decorate(block) {
+  const blockName = 'v2-hero';
+
+  const picture = block.querySelector('picture');
+  const link = block.querySelector('a');
+  const isVideo = link ? isVideoLink(link) : false;
+  if (isVideo) {
+    const video = createElement('video', [`${blockName}__video`]);
+    video.muted = true;
+    video.autoplay = true;
+
+    const source = createElement('source', '', {
+      src: link.getAttribute('href'),
+      type: 'video/mp4',
+    });
+    video.appendChild(source);
+    block.prepend(video);
+    link.remove();
+
+    // prevent video to reproduce more than 3s
+    video.addEventListener('timeupdate', checkVideoTime);
+  }
+
+  if (picture) {
+    const img = picture.querySelector('img');
+    img.classList.add(`${blockName}__image`);
+    if (picture.parentElement.tagName === 'P') {
+      picture.parentElement.remove();
+    }
+    block.prepend(picture);
+  }
+
+  const contentWrapper = block.querySelector(':scope > div');
+  contentWrapper.classList.add(`${blockName}__content-wrapper`);
+
+  // add Hero text variants to the corresponding content
+  const variantClasses = ['centered', 'left', 'bottom', 'dark'];
+  variantClasses.forEach((variant) => {
+    if (block.classList.contains(variant)) {
+      if (variant !== 'dark') block.classList.remove(variant);
```

Comment:
I think that we should modify the variant that comes to the block like this:
`dark` -> `v2-hero--dark`
`left` -> `v2-hero--left`
and then the original variant should be removed from the block. This will give us a simple way to check if the variant is supported by the block. Just inspect the element and check if the variant was transformed to the `v2-hero--${variant-name}`, if it is not transformed that means the block is not supporting this variant.
Thanks to the class name transformation we can prevent some global styles conflict too.

Example:
* We have a block that supports 2 variants `black`, and `white`
* somebody creates a block in the editor with some variants: `black` and `pink`
Because the block expects only `black` as a variant the classes of the rendered block would be like this:
`v2-hero v2-here--black pink`
At first glance, you can see that the `pink` is not supported as a variant.

WDYT?
BTW I did something like this here: 
https://github.com/Netcentric/vg-macktrucks-com-rd/blob/12f09b5239c649e9e4deed0ee4fad484dd070b94/blocks/v2-testimonial/v2-testimonial.js#L45

Reply:
I agree. 👍🏻